installers/msi-language/StateDeploy/CustomAction.cs

```diff
@@ -250,21 +250,44 @@ private static ActionResult _installStateTool(Session session, out string stateT
             if (oldPath.Contains(stateToolInstallDir))
             {
                 session.Log("State tool installation already on PATH");
-                return ActionResult.Success;
             }
-
-            var newPath = string.Format("{0};{1}", stateToolInstallDir, oldPath);
-            session.Log(string.Format("updating PATH to {0}", newPath));
-            try
+            else
             {
-                Environment.SetEnvironmentVariable("PATH", newPath, EnvironmentVariableTarget.Machine);
+                var newPath = string.Format("{0};{1}", stateToolInstallDir, oldPath);
+                session.Log(string.Format("updating PATH to {0}", newPath));
+                try
+                {
+                    Environment.SetEnvironmentVariable("PATH", newPath, EnvironmentVariableTarget.Machine);
+                }
+                catch (Exception e)
+                {
+                    string msg = string.Format("Could not update PATH. Encountered exception: {0}", e.Message);
+                    session.Log(msg);
+                    new SecurityError().SetDetails(session, msg);
+                    return ActionResult.Failure;
+                }
             }
-            catch (Exception e)
+
+            session.Log("Running prepare step...");
+            string prepareCmd = " --output=json" + " _prepare";
+            string prepareOutput;
+            ActionResult prepareRunResult = ActiveState.Command.Run(session, stateToolPath, prepareCmd, out prepareOutput);
+            if (prepareRunResult.Equals(ActionResult.Failure))
             {
-                string msg = string.Format("Could not update PATH. Encountered exception: {0}", e.Message);
+                string msg = string.Format("`state _prepare` returned with error: {0}", prepareOutput);
                 session.Log(msg);
-                new SecurityError().SetDetails(session, msg);
-                return ActionResult.Failure;
+                RollbarReport.Critical(msg, session);
+
+                Record record = new Record();
+                var errorOutput = Command.FormatErrorOutput(prepareOutput);
+                record.FormatString = string.Format("state _prepare failed with error:\n{0}", errorOutput);
+
+                session.Message(InstallMessage.Error | (InstallMessage)MessageBoxButtons.OK, record);
+                return prepareRunResult;
```

Comment:
```suggestion
                return ActionResult.Failure;
```

Be explicit

Reply:
I'm committing the suggestion, but I was trying to be cautious in the case that there was other data managed/propagated by that instance of the result.